lcdtest.py

```diff
@@ -102,67 +137,77 @@ def print_usage():
+        address = int(options_pop('addr'), 16)
+        port = int(options_pop('port', '1'))
+        try:
+            lcd = i2c.CharLCD(i2c_expander, address, port=port, charmap=charmap, cols=cols,
                           rows=rows, expander_params=options)
+        except IOError:
+            print_usage('IOError: Usually caused by the wrong i2c address/port or device not connected properly')
     elif lcdmode == 'gpio':
         if len(sys.argv) < 8:
             print_usage()
+
         # gpio options
-        data = options.pop('data')
-        rs = int(options.pop('rs'))
-        e = int(options.pop('e'))
-        rw = int(options.pop('rw'))
-        bl = options.pop('bl', None)
-        if bl is not None:
+        mode = options_pop('mode')
+        if mode == 'BCM':
+            numbering_mode = GPIO.BCM
+        elif mode == 'BOARD':
+            numbering_mode = GPIO.BOARD
+        else:
+            print_usage('Invalid GPIO numbering mode: mode=%s , must be either BOARD or BCM' % mode)
+
+        data = options_pop('data')
+        rs = int(options_pop('rs'))
+        e = int(options_pop('e'))
+
+        rw = options_pop('rw', None)
+        if rw == 'None' or rw is None:
+            rw = None
+        else:
+            rw = int(rw)
+
+        bl = options_pop('bl', None)
+        if bl == 'None' or bl is None:
+            bl = None
+        else:
             bl = int(bl)
+
         # Parse data pins into a list
         pins_data = {}
-        if sys.version_info > (3, 0):
-            pins_data = data.split(sep=',')
-            print(pins_data)
-        else:
-            pins_data = data.split(',')
+        pins_data = data.split(',')
         # Convert data pins to int
-        for i in range(len(pins_data)):
-            pins_data[i] = int(str(pins_data[i]))
-        print(pins_data)
+        pins_data = [int(pin) for pin in pins_data]
         lcd = gpio.CharLCD(pin_rs=rs, pin_rw=rw, pin_e=e, pins_data=pins_data,
-                       pin_backlight=bl, numbering_mode=mode, cols=cols, rows=rows,
+                       pin_backlight=bl, numbering_mode=numbering_mode, cols=cols, rows=rows,
                        charmap=charmap)
-        pass
     else:
-        print_usage()
+        print_usage('Connection type ' + lcdmode + ' is not supported. Must be either i2c or gpio')
 
     # Run selected test
     if test == 'show_charmap':
         show_charmap.run(lcd, rows, cols)
-    elif test == 'format_test':
-        if ((cols == 20) and (rows == 4)):
-            format_test_20x4.run(lcd)
-        elif ((cols == 16) and (rows == 2)):
-            format_test_16x2.run(lcd)
+    elif test == 'testsuite':
+        if cols == 20 and rows == 4:
+            testsuite_20x4.run(lcd)
+        elif cols == 16 and rows == 2:
+            testsuite_16x2.run(lcd)
         else:
-            print_usage()
+            print_usage(str(cols) + 'x' + str(rows) + ' displays not supported in this test.')
```

Comment:
Again, as noted above, it's easier to use string formatting:

```python
print_usage('%dx%d displays not supported in this test.' % (cols, rows))
```

There are some other instances of concatenation in this part of the code that could also be simplified like that.